runtime/compiler/env/VMJ9.cpp

```diff
@@ -2347,6 +2350,113 @@ TR_J9VMBase::markHotField(TR::Compilation * comp, TR::SymbolReference * symRef,
    }
 
 
+/**
+ * Report a hot field if the JIT has determined that the field has met appropriate thresholds to be determined a hot field. 
+ * Valid if dynamicBreadthFirstScanOrdering is enabled.
+ *
+ * @param reducedCpuUtil normalized cpu utilization of the hot field for the method being compiled
+ * @param clazz pointer to the class where a hot field should be added
+ * @param fieldOffset value of the field offset that should be added as a hot field for the given class
+ * @param reducedFrequency normalized block frequency of the hot field for the method being compiled
+ */
+void
+TR_J9VMBase::reportHotField(int32_t reducedCpuUtil, J9Class* clazz, U_8 fieldOffset,  U_32 reducedFrequency)
```

Comment:
High-level question - shouldn't this logic live in the GC with the logic that declares the data structures and the logic which reads the data structures? We should keep the concerns separated so perhaps moving this function into the GC and just calling the GC from this method would be more appropriate since these are internal details not owned by the JIT and we have traditionally kept the code separate to make sure we respect encapsulation and so the JIT doesn't have to change if you have to change something about the internal data representation in the GC.

Reply:
Yes I believe it is correct for this logic not live in the JIT. As this is purely modifying VM structures I will move the logic here to the VM side.